src/components/AppBar/AppBar.vue

```diff
@@ -120,67 +120,74 @@
   </v-app-bar>
 </template>
 
-<script>
+<script lang="ts">
+import { computed, defineComponent } from '@vue/composition-api';
+
 import {
-  cookiesEnabled, loggedIn, insideIFrame, publishRest,
+  cookiesEnabled as cookiesEnabledFunc,
+  loggedIn as loggedInFunc,
+  insideIFrame as insideIFrameFunc,
+  publishRest,
 } from '@/rest';
 import { dandiAboutUrl, dandiDocumentationUrl, dandiHelpUrl } from '@/utils/constants';
 import UserMenu from '@/components/AppBar/UserMenu.vue';
 
-export default {
+export default defineComponent({
   name: 'AppBar',
-  components: {
-    UserMenu,
-  },
-  data() {
-    return {
-      dandiAboutUrl,
-      dandiDocumentationUrl,
-      navItems: [
-        {
-          text: 'Welcome',
-          to: 'home',
-        },
-        {
-          text: 'Public Dandisets',
-          to: 'publicDandisets',
-        },
-        {
-          text: 'My Dandisets',
-          to: 'myDandisets',
-          if: loggedIn,
-        },
-        {
-          text: 'About',
-          to: dandiAboutUrl,
-          external: true,
-        },
-        {
-          text: 'Documentation',
-          to: dandiDocumentationUrl,
-          external: true,
-        },
-        {
-          text: 'Help',
-          to: dandiHelpUrl,
-          external: true,
-        },
-      ],
-    };
-  },
-  computed: {
-    loggedIn,
-    insideIFrame,
-    cookiesEnabled,
-    returnObject() {
-      const { name, query, params } = this.$route;
+  components: { UserMenu },
+  setup(props, ctx) {
+    const returnObject = computed(() => {
+      const { name, query, params } = ctx.root.$route;
       return JSON.stringify({ name, query, params });
-    },
-  },
-  methods: {
-    login() {
+    });
+
+    const cookiesEnabled = computed(cookiesEnabledFunc);
+    const loggedIn = computed(loggedInFunc);
+    const insideIFrame = computed(insideIFrameFunc);
```

Comment:
Probably out of scope for this PR, but we could probably just import `computed` in `rest.ts` and define these as `computed(() => ...)` instead of `() => ...`, which saves you having to do this song and dance in every component that needs to use them.

Reply:
I agree that would be better. I'll leave that for a future PR since that would affect a lot of other components too.

Reply:
Filed #909 for this.